vulnerabilities/tests/test_import_runner.py

```diff
@@ -20,260 +20,89 @@
         affected_packages=[
             AffectedPackage(
-                vulnerable_package=PackageURL(name="mock-webserver", type="pypi", version="1.2.33"),
-                patched_package=PackageURL(name="mock-webserver", type="pypi", version="1.2.34"),
+                package=PackageURL(type="pypi", name="dummy package"),
+                affected_version_range=VersionRange.from_string("vers:pypi/>=1.0.0|<=2.0.0"),
             )
         ],
+        references=[Reference(url="https://example.com/with/more/info/CVE-2020-13371337")],
+        date_published=datetime.now(timezone.utc),
     )
 ]
 
 
-def make_import_runner(added_advs=None, updated_advs=None):
-    added_advs = added_advs or []
-    updated_advs = updated_advs or []
-
-    importer = MockImporter(
-        data_source=MockDataSource(2, added_advs=added_advs, updated_advs=updated_advs)
-    )
+class DummyImporter(Importer):
+    spdx_license_expression = "dummy license"
 
-    return ImportRunner(importer, 5)
+    def advisory_data(self):
+        return ADVISORY_DATAS
 
 
-def test_ImportRunner_new_package_and_new_vulnerability(db):
-    runner = make_import_runner(added_advs=ADVISORIES)
-
+def test_import_runner(db):
+    runner = ImportRunner(DummyImporter)
     runner.run()
+    advisories = models.Advisory.objects.all()
+    advisory_datas = [x.to_advisory_data() for x in advisories]
+    assert advisory_datas == ADVISORY_DATAS
 
-    assert runner.importer.last_run is not None
-    assert runner.importer.saved
-
-    assert models.Package.objects.all().count() == 2
-    packages = models.Package.objects.filter(name="mock-webserver")
-
-    if packages[0].version == "1.2.33":
-        impacted_package, resolved_package = packages[0], packages[1]
-    else:
-        impacted_package, resolved_package = packages[1], packages[0]
-
-    assert models.Vulnerability.objects.count() == 1
-    assert models.VulnerabilityReference.objects.count() == 1
-    assert models.PackageRelatedVulnerability.objects.count() == 1
 
-    assert impacted_package.vulnerable_to.count() == 1
-    assert resolved_package.resolved_to.count() == 1
+def test_process_advisories_with_no_advisory(db):
+    process_advisories([], "")
+    assert 0 == models.Advisory.objects.count()
 
-    vuln = impacted_package.vulnerabilities.first()
-    assert vuln.vulnerability_id == "CVE-2020-13371337"
 
-    vuln_refs = models.VulnerabilityReference.objects.filter(vulnerability=vuln)
-    assert vuln_refs.count() == 1
-    assert vuln_refs[0].url == "https://example.com/with/more/info/CVE-2020-13371337"
+def test_process_advisories_with_advisories(db):
+    process_advisories(ADVISORY_DATAS, "test_importer")
+    advisories = models.Advisory.objects.all()
+    advisory_datas = [x.to_advisory_data() for x in advisories]
+    assert advisory_datas == ADVISORY_DATAS
 
 
-def test_ImportRunner_existing_package_and_new_vulnerability(db):
```

Comment:
Could this test be kept somehow? to valid that we never will allow twice the same set of imported data?

Reply:
This is done here
https://github.com/nexB/vulnerablecode/blob/c62f81f84c272686be972acebe15e2d2d6246e85/vulnerabilities/tests/test_import_runner.py#L80-L86